### DISCUSSION TITLE
fabric-lib-scrollbars

### DISCUSSION THREAD

**Discussion by zhe-he** [fabricjs/fabric.js#10371] [Show and tell]
I created a scrollbar using the canvas, where you can customize the scrollbar's stroke width, stroke color, fill color, minimum size, and more. Since the canvas must support zooming to enable the scrollbar, a mousewheel event is also provided. You can configure the maximum and minimum zoom levels.

Here’s a demo:
[Demo Access](https://codesandbox.io/p/devbox/t4kpd3?file=%2Fsrc%2Ftest.tsx%3A20%2C24)
[Alternative Access](https://stackblitz.com/edit/fabric-lib-scrollbars?file=src%2Ftest.tsx&terminal=dev)

### how to use
```ts
import { Scrollbars } from "./lib/scrollbars";
import { makeMouseWheel } from "./lib/scrollbars/util";

const mousewheel = makeMouseWheel(canvas);
canvas.on("mouse:wheel", mousewheel);
const scrollbar = new Scrollbars(canvas);

// dispose
// scrollbar.dispose();
// canvas.off("mouse:wheel", mousewheel);
```

**Comment by zhe-he** [comment]
https://github.com/user-attachments/assets/75f69359-9c96-4016-ab03-e0cf43dd3abf

Because the video has been compressed, it appears somewhat blurry

**Comment by m1rn** [comment]
@zhe-he scrollbar seems to scroll endlessly with no limit. How can I fix this problem?

**Reply by zhe-he** [reply]
Drawing on the canvas is boundless, so the scrollbars are also infinite.

If you want to implement your bounded functionality, you can make the canvas have boundaries—for example, by overriding the setViewportTransform method to restrict it within a certain range.
```ts
const canvas = new Canvas();

canvas.setViewportTransform = function (vpt) {
    const vpt2 = vpt.slice(0);
   //  if (vpt2[4] > xxx) xxx
   //  xxxx
    canvas.prototype.setViewportTransform.call(this, vpt2);
}
```

**Reply by m1rn** [reply]
Thanks a lot for your reply. Your idea really helped me, I really appreciate it.

**Comment by manisps2003-lang** [comment]
It’s a lightweight scrollbar module designed for the Fabric.js canvas framework. Instead of relying on browser-level scroll behavior, this module lets you embed fully customizable scrollbars directly within the canvas—making them look and feel like built-in UI elements.
